alloc.go

```diff
@@ -164,9 +176,6 @@ func (a *allocator) fileCommitPrepare(
 ) {
 	st.tx = tx
 	st.updated = forceUpdate || tx.Updated()
-	if st.updated {
-		a.MetaAllocator().FreeRegions(tx, a.freelistPages)
-	}
```

Comment:
Note: The 'freeing' of pages has been moved to `commitPrepareAlloc` in `tx.go`. This way `commitPrepareAlloc` mimics `commitPrepareWAL`, doing also the memory management and other side effectful operations. The `fileCommitPrepare` methods on the wal/allocator object will only compute meta information into the commit states, but are mostly side affect free by now (no operations creating IO or allocating/freeing pages).